services/order.py

```diff
@@ -0,0 +1,32 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> Order:
+    user = get_user_model().objects.get(username=username)
+    with transaction.atomic():
+        order = Order.objects.create(user=user,)
+        if date:
+            order.created_at = date
+
+        for ticket in tickets:
+            Ticket.objects.create(movie_session_id=ticket["movie_session"],
+                                  order=order,
+                                  row=ticket["row"],
+                                  seat=ticket["seat"]
+                                  )
+        order.save(update_fields=["created_at"] if date else None)

```

Comment:
You are saving the `order` instance multiple times which is against the DRY principle. You should call the `save()` method only once if it's possible. Try to organize your code in such a way that you only need to call `save()` once, after all changes to the instance have been made.